### DISCUSSION TITLE
Querying data from the client

### DISCUSSION THREAD

**Discussion by ElierHG** [internoapp/pocketblocks#51] [General]
Hello Andres. Thank you for your help the other day with getting started with the repo. I followed the instructions you provided, and I was able to get the dev setup working.

I have a question about client app, how do I perform queries? I notice the JS runtime does not have access to fetch. 
For example, let say I want to hit a URL when user clicks a button, what are my options? 

I found in the [docs](https://pocketblocks.pedroza.dev/miscellaneous/data-sources) that you decided not to include the Sources/Query library because the integration with Pocketbase was the primary goal. That means, querying data from the Pocketbase is easy and is done using **window.pb**. But, what if we create a custom Pocketbase route, is there an existing way we can hit that route if fetch is not available? Extending Pocketbase with routes will be useful in case there is a need to get data from an external source or trigger automation in n8n or cloud functions.

**Comment by ElierHG** [comment]
I found in **client/packages/openblocks-core/src/eval/utils/evalScript.tsx** where the fetch function is being disabled, so I commented line 11 to enable it temporarily. It seems to be working I can now make fetch requests, but I don't know if this breaks anything else in the app.

**Comment by pedrozadotdev** [comment]
I answered you in #52. You can also do what you said above, but it can create a security breach.

**Reply by ElierHG** [reply]
Thanks for the reply. 👍

I'm sure it was designed to be disabled for a reason. Anyways, most queries to external services need to be authenticated so I'll use what you suggested in #52.  Thanks!
